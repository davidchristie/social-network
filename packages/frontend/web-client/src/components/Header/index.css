.Header .Container {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: space-between;
}

.Header .items {
  display: flex;
}

.Header .items > *:not(:first-child) {
  margin-left: 10px;
}

.Header .logo {
  font-size: 1.4em;
  text-decoration: none;
}
