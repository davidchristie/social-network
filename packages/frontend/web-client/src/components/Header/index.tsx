import { Avatar, ButtonLink, Container, Topbar } from "design-system";
import { AccountMenu, ProfileSearch } from "domain-model";
import { Authenticated, Unauthenticated } from "domain-model";
import React from "react";
import { Link } from "react-router-dom";
import GitHub from "./GitHub.png";
import "./index.css";

export default class Header extends React.Component {
  public render () {
    return (
      <Topbar className="Header">
        <Container>
          <Link
            className="logo"
            to="/"
          >
            Social Network
          </Link>
          {this.renderMenuItems()}
        </Container>
      </Topbar>
    );
  }

  private renderMenuItems () {
    return (
      <div className="items">
        <Authenticated>
          <ProfileSearch />
          <AccountMenu />
        </Authenticated>
        <Unauthenticated>
          <ButtonLink to="/login">
            Login
          </ButtonLink>
          <ButtonLink to="/signup">
            Signup
          </ButtonLink>
        </Unauthenticated>
        <a href="https://github.com/davidchristie/social-network">
          <Avatar image={GitHub} />
        </a>
      </div>
    );
  }
}
