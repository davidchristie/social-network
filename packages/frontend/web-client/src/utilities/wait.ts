export default function wait (delay: number): Promise<void> {
  return new Promise(resolve => window.setTimeout(resolve, delay));
}
