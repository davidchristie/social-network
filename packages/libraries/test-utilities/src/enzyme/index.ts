export { default as beforeEachWaitForUpdate } from "./beforeEachWaitForUpdate";
export { default as itContainsComponent } from "./itContainsComponent";
