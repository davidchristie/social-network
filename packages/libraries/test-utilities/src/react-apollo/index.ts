export { default as mountWithMockedResponses } from "./mountWithMockedResponses";
