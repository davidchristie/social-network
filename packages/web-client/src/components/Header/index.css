.Header {
  align-items: center;
  background: var(--theme-primary, white);
  color: var(--theme-primary-contrast, black);
  display: flex;
  justify-content: space-between;
  padding: 1em;
}
