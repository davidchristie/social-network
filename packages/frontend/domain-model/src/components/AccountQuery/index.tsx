import React from "react";
import { Query, QueryResult } from "react-apollo";
import { Account } from "../../generated/types";
import ACCOUNT_QUERY from "../../queries/Account";

export type Result = QueryResult<Account>;

interface Props {
  children: (result: QueryResult<Account>) => React.ReactNode;
}

const AccountQuery: React.StatelessComponent<Props> = ({ children }) => (
  <Query<Account>
    query={ACCOUNT_QUERY}
  >
    {children}
  </Query>
);

export default AccountQuery;
