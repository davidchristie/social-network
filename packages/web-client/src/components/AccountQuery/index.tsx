import React from "react";
import { Query, QueryResult } from "react-apollo";

import Account, {
  AccountData,
  AccountVariables
} from "../../queries/Account";

interface Props {
  children: (result: QueryResult<AccountData, AccountVariables>) => React.ReactNode;
}

const AccountQuery = ({ children }: Props) => (
  <Query<AccountData, AccountVariables>
    query={Account}
  >
    {children}
  </Query>
);

export default AccountQuery;
