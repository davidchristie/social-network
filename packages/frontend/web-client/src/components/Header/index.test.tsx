import { mount } from "enzyme";
import React from "react";
import { MockedProvider, MockedResponse } from "react-apollo/test-utils";
import { MemoryRouter } from "react-router-dom";

import Header from ".";
import AccountQuery from "../../queries/Account";
import wait from "../../utilities/wait";

function mountWithProviders<T extends React.ComponentType> (
  Component: T,
  mocks: ReadonlyArray<MockedResponse>
) {
  return mount(
    <MockedProvider mocks={mocks}>
      <MemoryRouter>
        <Component />
      </MemoryRouter>
    </MockedProvider>
  );
}

describe("Header component", () => {
  it("while loading", async () => {
    const mocks = [
      {
        request: {
          query: AccountQuery,
        },
      },
    ];
    const wrapper = mountWithProviders(Header, mocks);
    expect(wrapper.find(Header)).toMatchSnapshot();
  });

  it("when logged out", async () => {
    const mocks = [
      {
        request: {
          query: AccountQuery,
        },
        result: {
          data: {
            account: null,
          },
        },
      },
    ];
    const wrapper = mountWithProviders(Header, mocks);
    await wait(0);
    wrapper.update();
    expect(wrapper.find(Header)).toMatchSnapshot();
  });
});
