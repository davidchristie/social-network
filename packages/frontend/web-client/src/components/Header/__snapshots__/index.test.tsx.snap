// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<Topbar
  className="Header"
>
  <Container>
    <Link
      className="logo"
      replace={false}
      to="/"
    >
      Social Network
    </Link>
    <div
      className="items"
    >
      <withRouter(ProfileSearch) />
      <Authenticated>
        <AccountMenu />
      </Authenticated>
      <Unauthenticated>
        <ButtonLink
          to="/login"
        >
          Login
        </ButtonLink>
        <ButtonLink
          to="/signup"
        >
          Signup
        </ButtonLink>
      </Unauthenticated>
      <a
        href="https://github.com/davidchristie/social-network"
      >
        <Avatar$2
          image="GitHub.png"
        />
      </a>
    </div>
  </Container>
</Topbar>
`;
